CONTRIBUTING.md

```diff
@@ -0,0 +1,161 @@
+# Contributing
+
+- [Contributing](#contributing)
+  - [Overview](#overview)
+    - [Pull request rules](#pull-request-rules)
+  - [Development Procedure](#development-procedure)
+    - [Testing](#testing)
+    - [Pull Requests](#pull-requests)
+    - [Requesting Reviews](#requesting-reviews)
+    - [Updating Documentation](#updating-documentation)
+  - [Dependencies](#dependencies)
+  - [Protobuf](#protobuf)
+  - [Branching Model and Release](#branching-model-and-release)
+    - [PR Targeting](#pr-targeting)
+
+## Overview
+
+This document codifies rules that must be followed when contributing to
+Babylon node repository.
+
+### Pull request rules
+
+Every pull request must be easy to review. To make it possible:
+
+ 1. **Each pull request must do _one thing_**. It must be very clear what that
+    one thing is when looking at the pull request title, description, and linked
+    issues. It must also be very clear what value it ultimately aims to deliver,
+    and for which user(s).
+
+ 2. **Each pull request must be manageable in size**.
+    Self-contained pull requests that are manageable in size may target `main` directly.
+    Larger contributions though must be structured as a series of smaller pull requests
+    each building upon the previous one, all ideally tracked in a tracking issue
+    These pull requests must target a long-lived feature branch.
+    For details, see the [development procedure guidelines](#development-procedure).
+
+    **Note**: This does not necessarily apply to documentation-related changes or
+    automatically generated code (e.g. generated from Protobuf definitions). But
+    automatically generated code changes should occur within separate commits, so
+    they are easily distinguishable from manual code changes.
+
+## Development Procedure
+
+`main` must be stable, include only completed features and never fail `make test`, `make test-e2e`, or `make build/install`.
+
+Depending on the scope of the work, we differentiate between self-contained pull requests and long-lived contributions (features).
+
+**Self-contained pull requests**:
+
+* Fork the repo (core developers must create a branch directly in the Babylon repo),
+branch from the HEAD of `main`, make some commits, and submit a PR to `main`.
+* Follow branch name conventions: `{prefix}/branch-name`, where `prefix` is one of the
+standard [types](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json)
```

Comment:
shall we include author name as part of the branch name? e.g., `johndoe/feature/add-user-profile`. [ref](https://phoenixnap.com/kb/git-branch-name-convention)

Reply:
tbh for my taste branch names are getting a bit long (super subjective opinion). Are there any other benefits of this approach except of additional information ?

Reply:
Nope, just easy to track who is working on this branch. We can say this is optional. For someone who wants to add the author name, we recommend going with `johndoe/feature/add-user-profile` format

Reply:
hmm I think being consistent here is more important that any particular format tbh. So how about we copy exactly what Gaia is doing:
1. for pr's directly to `main` the name is `moniker/branch-name`, where `moniker` is github name
2. for larger pr's we create feature branch with name `feat/feature-branch-name` , and pr's to this branch must follow the naming `moniker/branch-name`